### ISSUE TITLE
Use sabre/xml for parsing

### ISSUE THREAD

**Issue by sagikazarmark** [wsdl2phpgenerator/wsdl2phpgenerator#223]
I found sabre/xml the best userland XML reader implementation out there. It makes it super easy to map classes to namespaces/elements.

**Comment by kasperg** [comment]
In wsdl2phpgenerator most of the XML parsing is handled by `SoapClient` and DOMDocument is used for additional information.

If you think it can be improved I would be happy to discus how and review pull requests.

**Comment by sagikazarmark** [comment]
I am thinking about if the SoapClient solution could completely be replaced by a sabre/xml solution. What in my mind is that we could create a generic WSDL library which either reads or writes a WSDL document. It could be used then for generating PHP classes or generating a WSDL document from PHP.

What I am really proposing is to create a library like that and use it in the generator. Would you be interested in it?

**Comment by kasperg** [comment]
Yes - I am interested in solutions that remove the dependency on the PHP SoapClient extension.

It has a number of limitations which also affect the code generation process. A primary concern would be that such a solution would either have to be SoapClient compatible or provide an alternate solution for performing the actual SOAP requests.

If you want to work on that then I would recommend that you look into [the work](https://github.com/goetas?tab=repositories) by @goetas. He does not use sabre/xml however.

**Comment by goetas** [comment]
https://github.com/goetas-webservices and https://github.com/goetas/xsd2php are good start points

**Comment by sagikazarmark** [comment]
Will look into them.

**Comment by goetas** [comment]
Hi @kasperg , 
this is a bit old topic, but can be still interesting to have a look to  https://github.com/goetas-webservices/soap-client  released right today.

A fully PHP based soap client, with code generation for wsdl, soap and xsd metadata. No ext-soap dependency, PSR7 compatible and multi HTTP clients are supported (thanks @sagikazarmark  for your work on php-http)
